ios/RCTTWFrameCaptureRenderer.m

```diff
@@ -0,0 +1,53 @@
+//
+//  RCTTWFrameCaptureRenderer.m
+//  react-native-twilio-video-webrtc
+//
+//  Created by Christopher Fields II on 6/6/23.
+//
+
+#import <Foundation/Foundation.h>
+#import "RCTTWFrameCaptureRenderer.h"
+#import "RCTTWImageUtils.h"
+#include <math.h>
+
+// https://github.com/twilio/twilio-video-ios/issues/184#issuecomment-828655941 - overall idea inspiration
+// https://github.com/twilio/video-quickstart-ios/pull/286/files - more in depth implementation?
+
+@implementation RCTTWFrameCaptureRenderer
+
+- (void)captureFrame{
+    NSLog(@"[rntwilio] capturing the frame");
+    self.captureThisFrame = TRUE;
+}
+
+- (void)renderFrame:(nonnull TVIVideoFrame *)frame {
+    if (!self.captureThisFrame) {
+        return;
+    }
+
+    self.captureThisFrame = FALSE; // make sure to revert the flag
+
+    // do frame image conversion and save to file system
+    CIImage *ciImg = [CIImage imageWithCVImageBuffer:frame.imageBuffer];
+    UIImage *rawImg = [UIImage imageWithCIImage:ciImg];
+    UIImage *img = [rawImg rotate: M_PI / 2]; // rotate by 90 degrees
+    NSData *imageData = UIImagePNGRepresentation(img);
+    NSURL *url = [NSFileManager.defaultManager URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask].firstObject;
+    NSUUID *uuid = [NSUUID UUID];
+    NSString *fileId = [uuid UUIDString];
+    NSString *filename = [NSString stringWithFormat:@"rntframe-%@.jpeg", fileId];
+    url = [url URLByAppendingPathComponent:filename];
+    BOOL didSave = [imageData writeToFile:url.path atomically:TRUE];
```

Comment:
❓ Doesn't `imageData` contain a PNG image rather than a JPEG? Since it's the return value from `UIImagePNGRepresentation(img)` above.

Reply:
Yeah this was probably just a goof on my part accepting the intellisense suggestions. I should have used this API
https://developer.apple.com/documentation/uikit/1624115-uiimagejpegrepresentation?language=objc

great catch 🎉